From: =?utf-8?q?Jonas_=C3=85dahl?= <jadahl@gmail.com>
Date: Tue, 15 Aug 2023 22:10:31 +0200
Subject: tests/test-runner: Add `toggle_overview` command

This will emit the `overlay-key` key which will be handled by the test
shell.

Forwarded: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3195
Applied-upstream: 43.8, commit:150cf9b4aae5514a6f991f26a1729b8b53018f73
---
 src/tests/test-runner.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/tests/test-runner.c b/src/tests/test-runner.c
index bf08304..f758db6 100644
--- a/src/tests/test-runner.c
+++ b/src/tests/test-runner.c
@@ -1175,6 +1175,15 @@ test_case_do (TestCase *test,
 
       meta_display_focus_default_window (display, timestamp);
     }
+  else if (strcmp (argv[0], "toggle_overview") == 0)
+    {
+      MetaDisplay *display = meta_context_get_display (test->context);
+
+      if (argc != 1)
+        BAD_COMMAND ("usage: %s", argv[0]);
+
+      g_signal_emit_by_name (display, "overlay-key", 0);
+    }
   else
     {
       BAD_COMMAND("Unknown command %s", argv[0]);
